RELEASE.md

```diff
@@ -0,0 +1,35 @@
+# dbt-utils releases
+
+## When do we release?
+There's a few scenarios that might prompt a release:
+
+| Scenario                                   | Release type |
+|--------------------------------------------|--------------|
+| New functionality¹                         | minor        |
+| Breaking changes to existing macros        | minor        |
+| Fixes to existing macros                   | patch        |
+| dbt minor release with no breaking changes | patch        |
+| dbt minor release with breaking changes    | minor        |
+
+¹New macros were previously considered patch releases — we have brought them up to minor releases to make versioning for dependencies clearer.
+
+## Branching strategy
+
+At any point, there should be two long-lived branches:
+- `main`: This reflects the most recent release of dbt-utils
```

Comment:
I was thinking, "Shouldn't this be `dev/0.x-1.0` (currently `dev/0.5.0`)? But I think having a safe, clear, main branch to PR against for small fixes or README updates is going to be much more user-friendly.